rayon-core/src/thread_pool/mod.rs

```diff
@@ -80,6 +80,43 @@ impl ThreadPool {
     /// thread-local data from the current thread will not be
     /// accessible.
     ///
+    /// # Warning: execution order
+    ///
+    /// If the current thread is part of a different thread pool, it will try to
+    /// keep busy while the `op` completes in its target pool, similar to
+    /// calling [`ThreadPool::yield_now()`] in a loop. Therefore, it may
+    /// potentially schedule other tasks to run on the current thread in the
+    /// meantime. For example
+    ///
+    /// ```rust
+    /// # use rayon_core as rayon;
+    /// use rayon::iter::{IntoParallelIterator, ParallelIterator};
+    /// fn main() {
+    ///     rayon::ThreadPoolBuilder::new().num_threads(1).build_global().unwrap();
+    ///     let pool = rayon::ThreadPoolBuilder::default().build().unwrap();
+    ///     (0..3).into_par_iter().for_each(|_| {
```

Comment:
Oh, we can't use APIs from `rayon` proper while testing `rayon-core`. Maybe you can make the example with `join` instead?

Reply:
Foiled!  Rewritten to use `join()` as requested.